### ISSUE TITLE
Dealing with duplicate filenames

### ISSUE THREAD

**Issue by musjj** [Nudin/mplug#7]
How does your script currently deal with scripts sharing the same filename?
So for example, if the following repos exist:
https://github.com/foo/bar
https://github.com/foo1/bar
and they both contain `bar.lua`, how does your script resolve this issue when installing them to the user's MPV scripts directory?

**Comment by Nudin** [comment]
Good point! This currently would result in an error. This could be fixed by renaming the files to contain the plugin id (for example `pluginid-originafilename`).

**Comment by Koopzington** [comment]
According to the [mpv docs](https://mpv.io/manual/stable/#script-location) it also supports scripts located in a subfolder inside the scripts folder if they have a `main.x` inside them.

To deal with vendor-prefixing i'd suggest:

1. creating folder names along the lines of `platform-username-reponame`
2. check downloaded contents
    1. in case of a single script, rename it to `main.{ext}` and copy it into the folder
    2. in case of an existing `main.{ext}` file, copy all contents into the folder

Due to my lack of overall knowledge about mpv plugins though i don't know if this might break intercompatibility between separate scripts though.
